src/UI/RadioButton.elm

```diff
@@ -0,0 +1,96 @@
+module UI.RadioButton exposing (radioGroup)
+
+{-|
+
+
+# Grouped Elements
+
+@docs radioGroup
+
+-}
+
+import Element exposing (Attribute, Element, px)
+import Element.Background as Background
+import Element.Border as Border
+import Element.Events as Events
+import UI.Icon as Icon
+import UI.Internal.Palette as Palette
+import UI.Palette as Palette
+import UI.RenderConfig exposing (RenderConfig)
+import UI.Text as Text
+import UI.Utils.ARIA as ARIA
+import UI.Utils.Element as Element
+
+
+radioGroup : RenderConfig -> String -> (a -> msg) -> Maybe a -> List ( a, String ) -> Element msg
+radioGroup renderConfig label message selected buttons =
+    buttons
+        |> List.map
+            (\( id, buttonLabel ) ->
+                radioButton renderConfig
+                    (message id)
+                    buttonLabel
+                    (selected == Just id)
+            )
+        |> Element.column
+            (Element.spacing 8
+                :: (ARIA.toElementAttributes <| ARIA.roleRadioGroup label)
+            )
+
+
+radioButton : RenderConfig -> msg -> String -> Bool -> Element msg
+radioButton renderConfig message label state =
+    let
+        radioAttrs =
+            [ Element.width (px 26)
+            , Element.height (px 26)
+            , Border.color Palette.primary.middle
+            , Border.width 3
+            , Border.rounded 26
+            , Events.onClick message
+            , Element.pointer
+            ]
+                ++ (ARIA.toElementAttributes <| ARIA.rolePresentation)
```

Comment:
I think it's `role="radio"`  here

Reply:
Nope, according to the MDN/W3C:
The most extern column is the radiogroup
The row that has both icons and text is the radio (and has an aria-checked)
The image/icon/symbol/figure that represents the current state must be ignored as we already have it stated in aria-checked, so it's a presentation/none.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_radio_role

Reply:
that's true, my bad